src/Resources/app/administration/src/module/sw-in-app-purchases/component/sw-in-app-purchase-checkout/index.ts

```diff
@@ -70,13 +66,11 @@ export default Shopware.Component.wrapComponentConfig({
             this.state = 'loading';
 
             await Promise.all([
-                this.inAppPurchasesService.createCart(
-                    this.store.extension,
-                    this.store.entry.identifier
-                ),
-                this.inAppPurchasesService.getExtension(this.store.extension)
-            ]).then(([inAppPurchaseCart, extension]) => {
-                this.inAppPurchaseCart = inAppPurchaseCart;
+                this.inAppPurchasesService.getExtension(this.store.extension),
+                this.inAppPurchasesService.getPriceModels(this.store.extension, this.store.entry.identifier)
+            ]).then(([extension, purchase]) => {
+                this.purchase = purchase;
+                this.gtcAccepted = purchase.priceModels[0]?.conditionsType === null;
```

Comment:
Shouldn't this be part of checkout-overview and based on the selected price model?